### ISSUE TITLE
TinyImagenet数据集生成的问题

### ISSUE THREAD

**Issue by why3881** [TsingZ0/PFLlib#231]
首先很感谢作者优秀的工作和对开源仓库持续地维护！

最近在使用仓库生成TinyImagenet数据集时，发现了一些问题：在数据集划分过程中的输出和最终生成的config.json文件中，都可以明显地看出0号类别的数据量异常的大。具体来说，其他所有类别的总数据样本量均为500，而0号类别的数据样本量却是10500，整整多了10000。由于这个数据集是200个类别，训练集中每个类别是500个数据样本，而训练集和验证集均为每个类别50个样本（训练集中的样本不带标签信息，一般不使用），因此可以推测是对验证集的处理出现了一些问题。

在代码中打印相关类别信息：
`    unique, counts = np.unique(testset.targets.cpu().detach().numpy(), return_counts=True)`
`    print("验证集各类别样本数：", dict(zip(unique, counts)))`

运行后可以确认这个问题：

![Image](https://github.com/user-attachments/assets/f48eb239-1491-4ab0-9a19-58aabcb24193)

实际上，仓库数据集的划分逻辑是对训练集和测试集的数据先按类别进行合并，再按预设的比例重新进行划分。而这里错误地将验证集中的10000个数据样本都划分为了0号类别，也就是把验证集中所有类别的数据都错误地划分给了0号类别，又重新进行划分和训练，是严重不符合预期的。

排查了一下相关代码之后发现，这个问题是由TinyImagenet下载后的文件结构不匹配引起的，具体来说，在下载后的路径 ’dir_path/rawdata/tiny-imagenet-200/‘ 中有三个文件夹 train, test 和 val（这里我们只关心train 和 val），其中train文件夹下是200个不同类别的文件夹，再下面才是对应类别的图片；而val文件夹下则只有一个images文件夹，包含了所有10000张验证集的图片，以及一个 val_annotations.txt 文件，记录了每张图片与类别的映射信息。

![Image](https://github.com/user-attachments/assets/9ec1ba9a-4981-4379-b3f0-330f26bf9c25)

![Image](https://github.com/user-attachments/assets/96d0faf2-d857-4b55-863c-6e927db73d78)

而在现有的代码中，testset的路径使用的是`dir_path+'rawdata/tiny-imagenet-200/val/'`，在这个路径下只有images这个文件夹，因此在构建测试集时被错误地将全部图片都分类为了0号类别。

正确的处理方式应该是先对val文件夹进行重构，利用 val_annotations.txt 文件对里面的图片按类别分文件夹进行组织，然后再修改testset的路径为`dir_path+'rawdata/tiny-imagenet-200/val/images'`，这样就能正确地对验证集的图片进行分类了。

我借助LLM工具编写了相关的处理逻辑，并提交了PR，见 #230 

修改后的代码对val文件夹下的images文件夹进行了重构分类：

![Image](https://github.com/user-attachments/assets/8a7fc66b-95ec-48f4-9b87-3ffd1e350ea4)

验证集的类别也能够正确划分：

![Image](https://github.com/user-attachments/assets/d8cf4daf-f712-47b8-b9fc-4a1e106c8d5f)

麻烦作者在有空的时候审查一下相关代码，感谢，祝好！

**Comment by TsingZ0** [comment]
刚运行了一下最近的版本，确实有这个问题。但更早期版本的label 0数据也是500个，可能是版本更新时候出了点问题。为了跟早期版本对齐，也为了简单起见，我们只使用`tiny-imagenet-200/train`下面的数据。代码已更新

**Comment by why3881** [comment]
好的，那现在应该没什么问题了，感谢更新与回复
